internal/application/application.go

```diff
@@ -245,6 +245,27 @@ func deleteStagePVC(ctx context.Context, cluster *kubernetes.Cluster, appRef mod
 		PersistentVolumeClaims(deployments.TektonStagingNamespace).Delete(ctx, appRef.MakePVCName(), metav1.DeleteOptions{})
 }
 
+func PreviousStageId(ctx context.Context, cluster *kubernetes.Cluster, appRef models.AppRef) (string, error) {
+	wl := NewWorkload(cluster, appRef)
+
+	deployment, err := wl.Deployment(ctx)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return "", err
+		}
+		// App is inactive, no deployment, no workload, no id
+		return "", nil
+	}
+
+	workload := wl.Get(ctx, deployment)
```

Comment:
This `Get` is doing a little bit too much for what we need here (listing ingresses etc). Maybe it's better if implemented a method specifically for this use. E.g. `wl.GetStageID`. It should only get the needed field from the deployment (in one request against the kube api).

Reply:
Oh, what ?

 ... Do I see this correctly that `Get` is given the deployment in question as argument, and then still asks the kube cluster for it by searching via a List() call ?

Reply:
The only caller looks to be in the adjacent `application.go` and provides the deployment. This means the function can be massively simplified, it looks like.

Reply:
Changes done ... Specialized get for stage id. Simplified `Get`. Rebased to current main. Commit 96cb80e8c4d01c448661588dec6ca0263a8af82a